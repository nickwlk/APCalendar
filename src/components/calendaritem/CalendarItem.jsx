import React from "react";

//Calendar item
class CalendarItem extends React.Component{

constructor(props){
    super(props);
    this.updateNewItemValue = this.updateNewItemValue.bind(this);
    this.state={
        subject:"",text:"",time:"",date:""
    };
}
componentDidUpdate(){
    
    if (this.state.subject.length>0 && this.state.text.length>0 && this.state.time.length>0 && this.state.date.length>0){
        this.props.updateComplete(this.state.subject,this.state.text,this.state.time,this.state.date);
    }

}

    updateNewItemValue(evt){
        
        switch(evt.target.id){
            case "NewItemSubject":
                    this.setState({subject:evt.target.value});       
                break;
            case "NewItemText":
                    this.setState({text:evt.target.value});
                    break;
            case "NewItemTime":
                    this.setState({time:evt.target.value});
                break;
            case "NewItemDate":
                    this.setState({date:evt.target.value});
                break;
            default:
                break;
        }
        
    }

render(){
   
    return(
        <div >
            <div className="col">
            <label >Subject:
                <div>
                <input className="newItemElement" onChange={this.updateNewItemValue} value={ this.props.selectedAppointment.subject} type="text" id="NewItemSubject"></input>         
                </div>
            </label>
            </div>
            <label> Notes:
            <div>
                
                <textarea className="newItemElement" onChange={this.updateNewItemValue} value={this.props.selectedAppointment.text} rows="5" id="NewItemText"></textarea>
            </div>
            </label>
            <div>
                <label htmlFor="NewItemTime">Time:
                <div><input className="newItemElement" onChange={this.updateNewItemValue} value={this.props.selectedAppointment.time} type="time" id ="NewItemTime"></input></div></label>
                </div>
            <div>
                <label htmlFor="NewItemDate">Date:
                <div>
                <input className="newItemElement" onChange={this.updateNewItemValue} value={this.props.selectedAppointment.date} type="date" id ="NewItemDate"></input></div>
                </label>
            </div>
        </div>
    )
}

}
export default CalendarItem;