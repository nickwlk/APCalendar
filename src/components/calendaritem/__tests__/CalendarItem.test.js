import React from 'react';
import ReactDOM from 'react-dom';

import CalendarItem from '../CalendarItem';
import {cleanup} from "@testing-library/react";
import TestRenderer from 'react-test-renderer';

const testAppointment= {subject:'test',text:'test',time:'12:00',date:'2019/11/01'};

afterEach(cleanup);

it("renders without error",() =>{

  const div = document.createElement("div");
  ReactDOM.render(<CalendarItem selectedAppointment={testAppointment}></CalendarItem>,div);
})

it("matches snapshot",() =>{  
  const testAppointment= {subject:'test',text:'test',time:'12:00',date:'2019/11/01'};
    const tree = TestRenderer.create(<CalendarItem selectedAppointment={testAppointment}></CalendarItem>).toJSON();   
      
    expect(tree).toMatchSnapshot();
  });

