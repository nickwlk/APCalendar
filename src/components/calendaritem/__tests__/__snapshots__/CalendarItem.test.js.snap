// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="col"
  >
    <label>
      Subject:
      <div>
        <input
          className="newItemElement"
          id="NewItemSubject"
          onChange={[Function]}
          type="text"
          value="test"
        />
      </div>
    </label>
  </div>
  <label>
     Notes:
    <div>
      <textarea
        className="newItemElement"
        id="NewItemText"
        onChange={[Function]}
        rows="5"
        value="test"
      />
    </div>
  </label>
  <div>
    <label
      htmlFor="NewItemTime"
    >
      Time:
      <div>
        <input
          className="newItemElement"
          id="NewItemTime"
          onChange={[Function]}
          type="time"
          value="12:00"
        />
      </div>
    </label>
  </div>
  <div>
    <label
      htmlFor="NewItemDate"
    >
      Date:
      <div>
        <input
          className="newItemElement"
          id="NewItemDate"
          onChange={[Function]}
          type="date"
          value="2019/11/01"
        />
      </div>
    </label>
  </div>
</div>
`;
